control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3494,3 +3502,239 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
+}
+
+func awsSecurityGroupName(infraID string) string {
+	return fmt.Sprintf("%s-default-sg", infraID)
+}
+
+func createAWSDefaultSecurityGroup(ctx context.Context, ec2Client ec2iface.EC2API, infraID, vpcID string, additionalTags []hyperv1.AWSResourceTag) (string, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	// Determine VPC cidr
+	vpcResult, err := ec2Client.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		VpcIds: []*string{awssdk.String(vpcID)},
+	})
+	if err != nil {
+		logger.Error(err, "Failed to describe vpc", "vpcID", vpcID)
+		return "", fmt.Errorf("failed to describe vpc %s, code %s", vpcID, awsErrorCode(err))
+	}
+	if len(vpcResult.Vpcs) == 0 {
+		return "", fmt.Errorf("vpc %s not found", vpcID)
+	}
+	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
+	describeSGResult, err := ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{Filters: awsSecurityGroupFilters(infraID)})
+	if err != nil {
+		logger.Error(err, "Failed to list security groups")
+		return "", fmt.Errorf("cannot list security groups, code: %s", awsErrorCode(err))
+	}
+	sgID := ""
+	var sg *ec2.SecurityGroup
+	if len(describeSGResult.SecurityGroups) > 0 {
+		sg = describeSGResult.SecurityGroups[0]
+		sgID = awssdk.StringValue(sg.GroupId)
+	}
+	if sgID == "" {
+		// Create a security group if one is not found
+
+		tagKeys := sets.NewString()
+		var tags []*ec2.Tag
+		for _, tag := range additionalTags {
+			tagKeys.Insert(tag.Key)
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String(tag.Key),
+				Value: awssdk.String(tag.Value),
+			})
+		}
+		clusterKey := fmt.Sprintf("kubernetes.io/cluster/%s", infraID)
+		if !tagKeys.Has(clusterKey) {
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String(clusterKey),
+				Value: awssdk.String("owned"),
+			})
+		}
+		if !tagKeys.Has("Name") {
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String("Name"),
+				Value: awssdk.String(awsSecurityGroupName(infraID)),
+			})
+		}
+		if !tagKeys.Has("red-hat-managed") {
+			tags = append(tags, &ec2.Tag{
+				Key:   awssdk.String("red-hat-managed"),
+				Value: awssdk.String("true"),
+			})
+		}
+		createSGResult, err := ec2Client.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String(awsSecurityGroupName(infraID)),
+			Description: awssdk.String("default worker security group"),
+			VpcId:       awssdk.String(vpcID),
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags:         tags,
+				},
+			},
+		})
+		if err != nil {
+			logger.Error(err, "Failed to create security group")
+			return "", fmt.Errorf("failed to create security group, code: %s", awsErrorCode(err))
+		}
+		sgID = awssdk.StringValue(createSGResult.GroupId)
+
+		// Fetch just-created SG
+		describeSGInput := &ec2.DescribeSecurityGroupsInput{
+			GroupIds: []*string{awssdk.String(sgID)},
+		}
+		if err = ec2Client.WaitUntilSecurityGroupExistsWithContext(ctx, describeSGInput); err != nil {
```

Comment:
This sync call will have an impact in our SLO for HC time to go available.
That would be also avoided by https://github.com/openshift/hypershift/pull/2146/files#r1101039621

Reply:
Like I mentioned earlier, the KAS has to be functional first before we get into this code.